### ISSUE TITLE
[Web] [Accessibility][11yMAS][Name, Role and Value] [MAS4.1.2] [Incorrect role is defined for Search edit field.]

### ISSUE THREAD

**Issue by DasaradhaKamjula** [OfficeDev/script-lab#676]
**User Experience:**
User who rely on AT will not get proper information about "Search” if role is not defined properly which provides the purpose of control.
 
**Test Environment:**
OS Build: 16299.19
Browser: Microsoft Edge 41.16299.15.0
Tools Used: Developer Tool

**Attachments:**
 ![mas4 1 2_incorrect role for search](https://user-images.githubusercontent.com/33054846/32373415-2c6f0c3a-c0bf-11e7-9f62-c29c2831161d.png)
![mas4 1 2_incorrect role for menu](https://user-images.githubusercontent.com/33054846/32373456-543f3d98-c0bf-11e7-9e64-2388fe4066a6.png)
[MAS4.1.2 IncorrectRole.zip](https://github.com/OfficeDev/script-lab/files/1456680/MAS4.1.2.IncorrectRole.zip)

**Repro Steps:**
1. Open the URL: http://scriptlabdocsdemo.azurewebsites.net/Office%20Dev%20Center%20-%20Docs%20-%20Build%20your%20first%20Excel%20add-in.html .
2. Navigate to Search Field and open Developer tool.
3. Verify Role is defined properly for search field or not. 
 
**Current Behavior:** 
Role is defined as “Search” for “search field” which will confuse screen reader user .

**Expected Behavior:**
Role should be defined as text box as it is a edit field which will helps user to search.

**Note:** 
Same Issue is reproducible for following controls in the Header:
-->Incorrect role is defined for “Explore”, ”Resources” and “Documentation” control as menu items which will confuse the User . Expected is, Role should be defined as “Menu” as those controls are having “Menu Items” inside them.
-->Incorrect role is defined for "Getting Started" and "Code Samples" as Menu Item which will confuse the user. Expected is, Role should be defined as Links for them as they are redirecting to new page.
-->Same issue is repro for the [MAC + safari][ Build info: OS Build: Mac OS Sierra 0.12.6 Browser: Safari 11.0.1] Incorrect role is defined for search edit field , Explore , Resource ,documentation.

**MAS Reference:**
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={248054a6-5e68-4771-9e1e-242fb5025730}
 
**Suggested Fix:**
https://microsoft.sharepoint.com/teams/msenable/mas/pages/mas-details.aspx?masid=4.1.2
